### PR TITLE
Implment Redux tutorial

### PR DESCRIPTION
**Pull request by queq1890**
# What
- This is a pullrequest of My Implementation on  [React Redux Tutorial](https://redux.js.org/basics/usagewithreact)
- you can create Article through src/components/Form.js
- Created Article list is shown through src/components/List.js

# Why
- To learn basic of React with Redux

# Problem
- I think I also need to study [Redux Saga](https://github.com/redux-saga/redux-saga) 
- So I'm gonna do [this tutorial](https://redux-saga.js.org/docs/introduction/BeginnerTutorial.html) from official document

### CODE REVIEW

File: src/js/components/Form.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import { addArticle } from "../actions/index";
+
+const mapDispatchToProps = dispatch => {
+  return {
+    addArticle: article => dispatch(addArticle(article))
+  };
+};
+
+class ConnectedForm extends Component {
+  constructor() {
```
Comment (nvtcp9x) [inline src/js/components/Form.js]: you can write follow this way for a shorter code and not need constructor anymore:
state = {
  title: ""
}

handleChange = (event) => {
}

....

File: src/js/components/Form.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    const { title } = this.state;
+    const id = uuidvl();
+    this.props.addArticle({ title, id });
+    this.setState({ title: "" });
+  }
+
+  render() {
+    const { title } = this.state;
+    return (
+      <form onSubmit={this.handleSubmit}>
```
Comment (nvtcp9x) [inline src/js/components/Form.js]: Consider use this library for a better form https://github.com/jaredpalmer/formik

File: src/js/components/Form.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import uuidvl from "uuid";
+import { addArticle } from "../actions/index";
+
+const mapDispatchToProps = dispatch => {
+  return {
+    addArticle: article => dispatch(addArticle(article))
+  };
+};
+
+class ConnectedForm extends Component {
```
Comment (nvtcp9x) [inline src/js/components/Form.js]: You can use PureComponent for better performance, It's okay to replace all Components in a project into PureComponents, as long as you don't mutate directly any state

File: .babelrc [hunk 4]
```diff
@@ -1,5 +1,4 @@
 {
-    "presets": [
-        "env", "react"
-    ]
-}
\ No newline at end of file
+  "presets": ["env", "react"],
+  "plugins": ["transform-object-rest-spread"]
```
Comment (nvtcp9x) [inline .babelrc]: You can use "stage-0" in presets instead of adding more plugins for babel

**Comment by queq1890** [comment]
@nvtcp9x 
Thank you for reviewing my codes! I will fix them and notice you when everything is done.

**Comment by nvtcp9x** [comment]
Redux-saga is a good library to learn, we also use it in order to manage side-effect such as call API, nice step (y)

File: .babelrc (hunk 4 above)
Comment (queq1890) [reply .babelrc]: fixed

File: src/js/components/Form.js (hunk 3 above)
Comment (queq1890) [reply src/js/components/Form.js]: modified. I didn't know about PureComponent so the concept feels interesting :)

File: src/js/components/Form.js (hunk 1 above)
Comment (queq1890) [reply src/js/components/Form.js]: fixed.

File: package.json [hunk 5]
```diff
@@ -21,6 +21,7 @@
   "devDependencies": {
     "babel-core": "^6.26.0",
     "babel-loader": "^7.1.4",
+    "babel-plugin-transform-object-rest-spread": "^6.26.0",
```
Comment (nvtcp9x) [inline package.json]: Not need this line anymore :D

File: src/js/index.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import React from "react";
+import { render } from "react-dom";
+import { Provider } from "react-redux";
+import store from "./store/index";
+import App from "./components/App";
```
Comment (nvtcp9x) [inline src/js/index.js]: you should replace the link './components/App' into 'components/App' with a litle bit configuration in webpack (resolve)
In the next projects if you use eslint you need also use eslint-import-resolver-webpack in order to help eslint to understand your configuration about resolved path of webpack

File: src/js/index.js [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import React from "react";
+import { render } from "react-dom";
+import { Provider } from "react-redux";
+import store from "./store/index";
```
Comment (queq1890) [inline src/js/index.js]: @nvtcp9x Fixed the codes you mentioned above.

I wonder if I should also resolve the paths from like `src/js/reducers` `src/js/actions` `src/js/constants` `src/js/store`. 

In real project, are these paths resolved like `import store from "store/index"` ?

Comment (nvtcp9x) [reply src/js/index.js]: It's not necessary because we just need resolve which path that we use many times in our projects

**Pull request merged**